src/Abstractions/NexusMods.Abstractions.Settings/APropertyValueContainer.cs

```diff
@@ -0,0 +1,62 @@
+using DynamicData.Binding;
+using JetBrains.Annotations;
+
+namespace NexusMods.Abstractions.Settings;
+
+/// <summary>
+/// Helper abstract class for <see cref="SettingsPropertyValueContainer"/>.
+/// </summary>
+[PublicAPI]
+public abstract class APropertyValueContainer<T> : AbstractNotifyPropertyChanged
+{
+    private T _previousValue;
+    private T _currentValue;
+    private bool _hasChanged;
+
+    private IEqualityComparer<T> _equalityComparer;
```

Comment:
What about the default value?
If it is not contained in the ValueContainer then the UI will not have access to it. 
I think we do want to know whether the current value is different from the default and have the ability to revert to default.

Reply:
Right, forgot to add that.